quasar-core/src/main/java/co/paralleluniverse/fibers/instrument/InstrumentClass.java

```diff
@@ -125,7 +130,10 @@ public void visitSource(String source, String debug) {
     }
 
     boolean hasSuspendableMethods() {
-        return methods != null && !methods.isEmpty();
+        return methodsSuspendable != null && !methodsSuspendable.isEmpty();
+    }
+    boolean hasNonSuspendableMethods() {
```

Comment:
That function name isn't quite correct, is it? This is not checking for generally non-suspendable methods, but specifically for synthetic methods that have a chance of being made suspendable. I think the name should reflect this - as there could be genuinely, user-defined non-suspendable methods that this check would not consider.